internal/pkg/workspace/workspace.go

```diff
@@ -515,26 +522,62 @@ func (ws *Workspace) Path() (string, error) {
 	return filepath.Dir(copilotDirPath), nil
 }
 
-// Rel returns the path relative to the workspace root.
-func (ws *Workspace) Rel(fullPath string) (string, error) {
+// RelWsRoot returns the path relative to the workspace root.
+// The input path is allowed to be either relative to the current
+// working directory or absolute.
+//
+// This is useful for storing a file path in configuration.
+//
+// Prefer RelCwd for displaying paths to users.
+func (ws *Workspace) RelWsRoot(path string) (string, error) {
 	copiDir, err := ws.copilotDirPath()
 	if err != nil {
 		return "", fmt.Errorf("get path to Copilot dir: %w", err)
 	}
+	fullPath, err := filepath.Abs(path)
+	if err != nil {
+		return "", fmt.Errorf("make path absolute: %w", err)
+	}
 	return filepath.Rel(filepath.Dir(copiDir), fullPath)
 }
 
+// RelCwd returns the path relative to the current working directory.
+// The input path is allowed to be either relative to the current
+// working directory or absolute.
+//
+// This is useful when displaying a file path to the user.
+//
+// You probably want to use RelWsRoot for most other cases.
+// In the common case (the user running commands from the workspace root)
+// these functions have the same behavior, which can be a dangerous
+// source of false assumptions because that is not guaranteed!
+func (ws *Workspace) RelCwd(fullPath string) (string, error) {
+	if !filepath.IsAbs(fullPath) {
+		// A non-absolute path must be relative to the current working
+		// directory, so just clean it and give it back.
+		return filepath.Clean(fullPath), nil
+	}
+
+	return filepath.Rel(ws.workingDir, fullPath)
+}
+
+// copilotDirPath tries to find the current app's copilot directory from the workspace working directory.
 func (ws *Workspace) copilotDirPath() (string, error) {
 	if ws.copilotDir != "" {
 		return ws.copilotDir, nil
 	}
-	// Are we in the application directory?
+	// Are we in the application's copilot directory already?
 	inCopilotDir := filepath.Base(ws.workingDir) == CopilotDirName
 	if inCopilotDir {
```

Comment:
I'm not sure about this change. Saying "a 'copilot' directory" sorta loses the intent conveyed in the original comment, the intent that this code is meant to determine whether we are in *the* single copilot directory for the application. While it is true that what the code *does* is that it detects if we are in a directory named copilot, in the case where we're in a directory that the user happened to name "copilot" (but is not the workspace's copilot folder) this code will incorrect, and someone who wants to figure out how to fix it would be guided by the *intent* in the comment.

How about something like this?

```suggestion
	// Are we in the application's copilot directory already?
	//
	// Note: This code checks for *any* directory named "copilot", but this might not work
	// correctly if we're in some subdirectory of the app that the user might have happened
	// to name "copilot". It's not clear if there's a good way to avoid that problem, though it
	// doesn't seem to be a terribly likely issue.
	if filepath.Base(ws.workingDir) == CopilotDirName {
```

Reply:
Sounds good, thanks Ryan!